### ISSUE TITLE
Building Asar broken with asar@0.14.4

### ISSUE THREAD

**Issue by markmcdowell** [electron-userland/electron-packager#892]
- [x] I have read the contribution documentation for this project.
- [x] I agree to follow the code of conduct that this project follows, as appropriate.
- [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

Hello! If you do a fresh install with `npm i electron-packager` it brings down asar@0.14.4.

Running `electron-packager . --asar` then fails with an ENOENT error as the file destination is no-longer correct: (it looks like the temp directory is used twice).

As a quick fix I fixed the version of asar:

`npm i asar@0.14.3 -D`

Error
```
  flora-colossus creating walker with rootModule=dist +0ms
  electron-packager Running asar with the options {} +14ms
fs.js:941
   binding.lstat(pathModule._makeLong(path));
          ^

Error: ENOENT: no such file or directory, lstat '%TMP%\electron-packager\win32-x64\app\resources\app\%TMP%\electron-packager\win32-x64\app\resources\app\main.js'
    at Object.fs.lstatSync (fs.js:941:11)
    at handleFile (D:\dev\trading-ui\electron-host\node_modules\asar\lib\asar.js:116:23)
    at next (D:\dev\trading-ui\electron-host\node_modules\asar\lib\asar.js:170:12)
    at Object.module.exports.createPackageFromFiles (D:\dev\trading-ui\electron-host\node_modules\asar\lib\asar.js:175:10)
    at D:\dev\trading-ui\electron-host\node_modules\asar\lib\asar.js:50:20
    at D:\dev\trading-ui\electron-host\node_modules\asar\lib\crawlfs.js:19:12
    at f (D:\dev\trading-ui\electron-host\node_modules\once\once.js:25:25)
    at Glob.<anonymous> (D:\dev\trading-ui\electron-host\node_modules\asar\node_modules\glob\glob.js:146:7)
    at emitOne (events.js:116:13)
    at Glob.emit (events.js:211:7)
    at Glob._finish (D:\dev\trading-ui\electron-host\node_modules\asar\node_modules\glob\glob.js:185:8)
    at done (D:\dev\trading-ui\electron-host\node_modules\asar\node_modules\glob\glob.js:172:12)
    at Glob._processGlobStar2 (D:\dev\trading-ui\electron-host\node_modules\asar\node_modules\glob\glob.js:614:12)
    at D:\dev\trading-ui\electron-host\node_modules\asar\node_modules\glob\glob.js:603:10
    at RES (D:\dev\trading-ui\electron-host\node_modules\inflight\inflight.js:31:16)
    at f (D:\dev\trading-ui\electron-host\node_modules\once\once.js:25:25)
````

**Comment by malept** [comment]
Can you please file a bug upstream in the [`asar` repository](https://github.com/electron/asar), if `asar` is broken? I cannot reproduce this on Linux:

1. `git clone https://github.com/electron/electron-quick-start && cd electron-quick-start`
2. `npm install && npm install --save-dev electron-packager`
3. `$(npm bin)/electron-packager . --asar`

**Comment by markmcdowell** [comment]
Sure, the same commands you ran on Linux fails on Windows.

**Comment by malept** [comment]
Perhaps it's a Windows-only problem in `asar` then?

**Comment by Appuls** [comment]
> As a quick fix I fixed the version of asar:
> 
> `npm i asar@0.14.3 -D`

I just ran into the same problem on Windows and your quick fix worked for me. Thanks!

**Comment by mrthorne** [comment]
Also ran into the same issue with Asar 0.14.4; forcing version 0.14.3 allows us to work around for now as well.

![image](https://user-images.githubusercontent.com/7717451/46310327-e5ac8600-c573-11e8-810c-413c641642b0.png)

**Comment by malept** [comment]
To people affected by this:

This is a confirmed bug with the `asar` module on Windows. Per the original post, there is a workaround to pin the `asar` version to the last known good version.

It would be appreciated if people refrained from saying "I ran into the same issue, the workaround worked", as it causes unnecessary notifications for people subscribing to this issue.

It would also be appreciated if someone could file a detailed bug report at the asar repository so its developers know about the regression.

**Comment by malept** [comment]
This should be fixed in asar 0.14.5. [Please let upstream developers know if it still doesn't work](https://github.com/electron/asar/issues/158), after you've verified that `asar` has indeed been updated to >= 0.14.5 for the Electron Packager installation you're using. A common mistake is updating the Packager dependencies for the project-specific installation, and then running the globally-installed Packager.
